### ISSUE TITLE
Progress bar does not show when initiating a login

### ISSUE THREAD

**Issue by emanuelen5** [makerspace/memberbooth#83]
Perhaps it should also say: "tag read" immediately when a tag has been sensed

**Comment by emanuelen5** [comment]
The progress bar still does not manage to get redrawn before the request is being sent. I managed to recreate the conditions without a reader by using a mocked reader (I had to chew on the code to make it show the behavior I wanted):

* In c22332d91dadec034a80458456e7504b6b4fa0b8 I skip the check if the reader is connected, and also induce a delay before returning the request data (press <kbd>A</kbd> when it says **Scan tag** to send the event). This does not work.

* In e35e4ec446275319fd9d0f6358bcf1c8cb8e6ebf I added a call to `Frame.update()` before retrieving the request data. This manages to trigger a *single* redraw. However, It does not animate the progress bar.

So I think the conclusion is that the main thread is used for drawing the screen, but we're keeping it busy while processing the request. We therefore need use another thread for handling the requests if we want it to keep drawing the screen.
